### ISSUE TITLE
MongoDB canned app not working - chronograf 1.8.8

### ISSUE THREAD

**Issue by BalzySte** [influxdata/chronograf#5762]
Hello!
I'm logging MongoDB statistics through Telegraf. That seems to work okay, data is available and I can plot it with custom queries. However, the mongodb canned app is not automatically detected and the template dashboard is not available in the hosts panel.

I've been searching a lot but the only related issues are 3-4 years old and have already been fixed.
Has anyone met the same issue? Any help is appreciated!

__Steps to reproduce:__
1. setup influxdb, chronograf, telegraf and mongodb with default configs
2. configure telegraf to log mongodb data (see conf below)
3. check chronograf dashboard

![image](https://user-images.githubusercontent.com/8466394/120918523-23fb8b00-c6b5-11eb-817e-ab4fc1c25394.png)

![image](https://user-images.githubusercontent.com/8466394/120918544-3c6ba580-c6b5-11eb-8f6c-4cda2bbd8a67.png)

__Expected behavior:__
mongodb app should appear in the "hosts panel" together with system and influxdb

__Actual behavior:__
Despite mongodb statistics being available, the mongodb app does not appear in host panel

__Environment info:__
Chronograf 1.8.8 - docker (telegraf:latest - 870ee38129f8), default config
Telegraf 1.18.3
InfluxDB v1.8.4 (git: 1.8 bc8ec4384eed25436d31045f974bf39f3310fa3c)

__Config:__
Telegraf config
``` conf
# Read metrics from one or many MongoDB servers
[[inputs.mongodb]]
  servers = ["mongodb://user:password@hostname:27017"]
```

**Comment by timhallinflux** [comment]
I believe you can attempt to import it through the Configuration | Connection | Update Connection -- it should give you the option to bring the MongoDB dashboard in.  I'm not sure that was ever integrated into the Host List page.

**Comment by sranka** [comment]
I can't reproduce it (having the latest nightly build of chronograf). MongoDB is offered when I update the connection ... and at the end, `mongodb' appears in the Apps columns of the hosts table:

![image](https://user-images.githubusercontent.com/16321466/122362333-55e3db80-cf58-11eb-83e5-aaa99b07dddf.png)

Is your `telegraf` database properly configured in your InfluxDB  connection?
![image](https://user-images.githubusercontent.com/16321466/122363278-1bc70980-cf59-11eb-9a33-acd51b813268.png)

UI detects host Apps using the following influxQL query

```
use telegraf;
show series from /^apache$|^consul_health_checks$|^consul_consul_fsm_register$|^consul_memberlist_msg_alive$|^consul_raft_state_candidate$|^consul_consul_http_GET_v1_health_state__$|^consul_raft_state_leader$|^consul_serf_events$|^cpu$|^disk$|^diskio$|^docker$|^docker_container_blkio$|^docker_container_net$|^elasticsearch_indices$|^haproxy$|^influxdb_database$|^influxdb_httpd$|^influxdb_queryExecutor$|^influxdb_write$|^kubernetes_node$|^kubernetes_pod_container$|^kubernetes_pod_network$|^kubernetes_system_container$|^system$|^mem$|^memcached$|^mesos$|^mongodb$|^mysql$|^net$|^netstat$|^nginx$|^nsq_channel$|^nsq_server$|^nsq_topic$|^phpfpm$|^ping$|^postgresql$|^processes$|^procstat$|^rabbitmq_node$|^redis$|^riak$|^varnish$|^win_cpu$|^win_mem$|^win_net$|^win_system$|^win_websvc$/ where time > now() - 10m
```

The `from` part the series query is programmatically generated from all the apps returned by http://localhost:8080/chronograf/v1/layouts. Is `mongodb` included in your case?

**Comment by BalzySte** [comment]
Thank you for both your feedbacks!

@timhallinflux as a matter of facts I had already tried what you suggested before posting the issue, I've checked again and I can confirm the the MongoDB Dashboard is not available in the list.

@sranka I've followed all the steps in your reply, and I think I've figured out where the issue is.

**TLDR:**  [http://localhost:8080/chronograf/v1/layouts](http://localhost:8080/chronograf/v1/layouts) does not return any **mongodb** app among the items

Going into detail, the Chronograf connection config is correct, and the result from the the SHOW SERIES query contains a **mongodb** series.

![image](https://user-images.githubusercontent.com/8466394/122655611-e0445f00-d153-11eb-8cef-05ca7361c46b.png)

On the other hand, the [/layouts](url) endpoint does not return any **mongodb** application.
This is a sorted list of all the **app** attributes 

```python
['apache', 'consul', 'consul_telemetry', 'consul_telemetry', 'consul_telemetry', 'consul_telemetry', 'consul_telemetry', 'docker', 'docker', 'docker', 'elasticsearch', 'haproxy', 'iis', 'influxdb', 'influxdb', 'influxdb', 'influxdb', 'kubernetes', 'kubernetes', 'kubernetes', 'kubernetes', 'memcached', 'mesos', 'mysql', 'nginx', 'nsq', 'nsq', 'nsq', 'phpfpm', 'ping', 'postgresql', 'rabbitmq', 'redis', 'riak', 'system', 'system', 'system', 'system', 'system', 'system', 'system', 'system', 'system', 'win_system', 'win_system', 'win_system', 'win_system']
```

I'm running the following official docker image:
```
chronograf          latest              f0eb4158cc69        5 weeks ago         195MB
```
which comes with Chronograf 1.8.8
```shell
$ docker run -it --rm chronograf chronograf --version
Chronograf 1.8.8 (git: 5c1aac9ebb63e2a91b0cf04d4b86202a7130920d)
```

**Comment by sranka** [comment]
It works fine when running 
```
docker run -it -p 8888:8888  chronograf
```

It is likely the case that `mongodb.json` in your docker's `/usr/share/chronograf/canned` is corrupt or missing. The Config section from `docker inspect <your chronograf container>` would help to identify the exact directory from which mongodb.json is read.

**Comment by sranka** [comment]
Independently on this issue, it is better to start with the latest chronograf docker image, which is now quay.io/influxdb/chronograf:1.8.10 . More details are available at https://portal.influxdata.com/downloads/

**Comment by BalzySte** [comment]
@sranka here is the 'Config' section from `docker inspect chronograf`

```json
        "Config": {
            "Hostname": "3ab31686277f",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "8888/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "HTPASSWD=/var/lib/chronograf/htpasswd",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "CHRONOGRAF_VERSION=1.8.8"
            ],
            "Cmd": [
                "--influxdb-url",
                "http://influxdb:8086",
                "--influxdb-username",
                "censored_username",
                "--influxdb-password",
                "censored_password"
            ],
            "Image": "chronograf",
            "Volumes": {
                "/var/lib/chronograf": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "/entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {}
        },
```

I've checked `mongodb.json` in `/usr/share/chronograf/canned/`, it exists and shows no difference with the one shipped in the official image you suggested. 
 
 I've also tested the **chronograf 1.8.10** image you suggested and here's the results:
 - MongoDB still does not appear in the **Host List** panel
 - `mongodb` app is still missing from the list provided by `http://hostname:8888/chronograf/v1/layouts`
 - I was able to add the MongoDB dashboard through the **Configuration** panel: *Connection configuration* -> *Dashboards*
 - The dashboard seems to work as expected and logged data from telegraf is displayed
 
 Thank you for your help, really appreciated!

**Comment by sranka** [comment]
Thank you Stefano for the required details and your investigation.

Adding the following environment variables will fix it in both versions

```sh
PROTOBOARDS_PATH=/usr/share/chronograf/protoboards
CANNED_PATH=/usr/share/chronograf/canned

# 1.8.8 works fine with
docker run -d -e PROTOBOARDS_PATH=/usr/share/chronograf/protoboards -e CANNED_PATH=/usr/share/chronograf/canned --name=chronograf188 -p 8889:8888 chronograf

# 1.8.10 works fine with
docker run -d -e PROTOBOARDS_PATH=/usr/share/chronograf/protoboards -e CANNED_PATH=/usr/share/chronograf/canned --name=chronograf18x -p 8890:8888 quay.io/influxdb/chronograf:1.8.10
```

TLDR, The root cause is that the default docker cmd is incomplete (1.8.8) and that the supplied custom cmd does not keep the default options (1.8.8 and 1.8.10).

I am about to fix this for chronograf 1.9
